cpp/src/join/join_common_utils.hpp

```diff
@@ -22,6 +22,7 @@
 #include <hash/concurrent_unordered_multimap.cuh>
 
 #include <cuco/static_multimap.cuh>
+#include <rmm/mr/device/polymorphic_allocator.hpp>
```

Comment:
Order includes nearest to farthest (top to bottom) and separate libraries with a blank line.

```suggestion
#include <rmm/mr/device/polymorphic_allocator.hpp>

#include <cuco/static_multimap.cuh>
```